wearApp/src/main/java/dev/johnoreilly/confetti/wear/auth/GoogleSignInPromptScreen.kt

```diff
@@ -0,0 +1,74 @@
+@file:OptIn(ExperimentalHorologistAuthComposablesApi::class,
+    ExperimentalHorologistComposeLayoutApi::class, ExperimentalHorologistBaseUiApi::class,
+    ExperimentalHorologistAuthUiApi::class, ExperimentalHorologistAuthUiApi::class
+)
+
+package dev.johnoreilly.confetti.wear.auth
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.wear.compose.material.Text
+import com.google.android.horologist.auth.composables.ExperimentalHorologistAuthComposablesApi
+import com.google.android.horologist.auth.composables.chips.SignInChip
+import com.google.android.horologist.auth.ui.ExperimentalHorologistAuthUiApi
+import com.google.android.horologist.auth.ui.common.screens.prompt.SignInPromptScreen
+import com.google.android.horologist.auth.ui.common.screens.prompt.SignInPromptViewModel
+import com.google.android.horologist.base.ui.ExperimentalHorologistBaseUiApi
+import com.google.android.horologist.base.ui.components.ConfirmationDialog
+import com.google.android.horologist.base.ui.components.StandardChipType
+import com.google.android.horologist.compose.layout.ScalingLazyColumnState
+import com.google.android.horologist.compose.navscaffold.ExperimentalHorologistComposeLayoutApi
+import dev.johnoreilly.confetti.R
+import org.koin.androidx.compose.getViewModel
+
+@Composable
+fun GoogleSignInPromptScreen(
+    navigateToGoogleSignIn: () -> Unit,
+    navigateUp: () -> Unit,
+    columnState: ScalingLazyColumnState,
+    modifier: Modifier = Modifier,
+    viewModel: SignInPromptViewModel = getViewModel()
+) {
+    var showAlreadySignedInDialog by rememberSaveable { mutableStateOf(false) }
+
+    SignInPromptScreen(
+        message = stringResource(id = R.string.google_sign_in_prompt_message),
+        onAlreadySignedIn = {
+            showAlreadySignedInDialog = true
+        },
+        columnState = columnState,
+        modifier = modifier,
+        viewModel = viewModel
+    ) {
+        item {
+            SignInChip(
+                onClick = {
+                    navigateToGoogleSignIn()
+                },
+                chipType = StandardChipType.Secondary
+            )
+        }
+    }
+
+    if (showAlreadySignedInDialog) {
```

Comment:
the logic [here](https://github.com/joreilly/Confetti/pull/295/files#diff-2e42dcaf3bfd1f6ea686706be36118e3da79a5879291bbd8761fc22254c5028bR51) seems to make sure this screen is not called when it's already signed, I guess it's just being cautious here?

Reply:
Yeah, I think this screen should work even if we decided to do this at startup.

Reply:
In the startup I'd say it would be better to just dismiss the screen instead of blocking the user journey with a dialog, but yeah, other scenarios might be appropriate to display the dialog to not look like the button is broken 👍